index.html

```diff
@@ -204,55 +204,1951 @@ <h3>
+              [=relevant global object=] of [=this=] using the [=HID device
+              task source=] to [=reject=] |promise| with a
+              "{{NotAllowedError}}" {{DOMException}} and abort these steps.
+              </li>
+              <li>Set [=this=].{{HIDDevice/[[state]]}} to `"opened"`.
+              </li>
+              <li>[=Queue a global task=] on the [=relevant global object=] of
+              [=this=] using the [=HID device task source=] to [=resolve=]
+              |promise| with `undefined`.
+              </li>
+            </ol>
+          </li>
+          <li>Return |promise|.
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h3>
+          <dfn>close()</dfn> method
+        </h3>
+        <p>
+          The {{HIDDevice/close()}} method steps are:
+        </p>
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].
+          </li>
+          <li>Set [=this=].{{HIDDevice/[[state]]}} to `"closing"`.
+          </li>
+          <li>Perform the following steps [=in parallel=]:
+            <ol>
+              <li>Invoke the operating system to close the HID device and
+              release any associated resources.
+              </li>
+              <li>Set [=this=].{{HIDDevice/[[state]]}} to `"closed"`.
+              </li>
+              <li>[=Queue a global task=] on the [=relevant global object=] of
+              [=this=] using the [=HID device task source=] to [=resolve=]
+              |promise| with `undefined`.
+              </li>
+            </ol>
+          </li>
+          <li>Return |promise|.
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h3>
+          <dfn>sendReport()</dfn> method
+        </h3>
+        <p>
+          The {{HIDDevice/sendReport()}} method steps are:
+        </p>
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].
+          </li>
+          <li>If [=this=].{{HIDDevice/[[state]]}} is not `"opened"`, [=reject=]
+          |promise| with a "{{InvalidStateError}}" {{DOMException}} and abort
+          these steps.
+          </li>
+          <li>Perform the following steps [=in parallel=]:
+            <ol>
+              <li>Invoke the operating system to send an output report with the
+              specified |reportId:octet| and |data:BufferSource|.
+              </li>
+              <li>If this report is a [=blocked report=], [=queue a global
+              task=] on the [=relevant global object=] of [=this=] using the
+              [=HID device task source=] to [=reject=] |promise| with a
+              "{{NotAllowedError}}" {{DOMException}} and abort these steps.
+              </li>
+              <li>If this fails for any reason, [=queue a global task=] on the
+              [=relevant global object=] of [=this=] using the [=HID device
+              task source=] to [=reject=] |promise| with a
+              "{{NotAllowedError}}" {{DOMException}} and abort these steps.
```

Comment:
This is a little bit of a weird error code. I would use `NetworkError`. I see that in Chromium there's a limitation because the IPC doesn't return a more expressive error code.

Reply:
Sounds good, switched to `NetworkError`